test/int/integration_suite_test.go

```diff
@@ -183,159 +197,162 @@ func defaultTimeouts() {
-		DeprecatedResourceWatcher:   watch.NewDeprecatedResourceWatcher(),
-		AtlasProvider:               atlasProvider,
-		GlobalPredicates:            globalPredicates,
-		ObjectDeletionProtection:    deletionProtection,
-		SubObjectDeletionProtection: false,
-	}).SetupWithManager(k8sManager)
-	Expect(err).ToNot(HaveOccurred())
-
-	err = (&atlasfederatedauth.AtlasFederatedAuthReconciler{
-		Client:                      k8sManager.GetClient(),
-		Log:                         logger.Named("controllers").Named("AtlasFederatedAuth").Sugar(),
-		DeprecatedResourceWatcher:   watch.NewDeprecatedResourceWatcher(),
-		GlobalPredicates:            globalPredicates,
-		EventRecorder:               k8sManager.GetEventRecorderFor("AtlasFederatedAuth"),
-		AtlasProvider:               atlasProvider,
-		ObjectDeletionProtection:    deletionProtection,
-		SubObjectDeletionProtection: false,
-	}).SetupWithManager(k8sManager)
-	Expect(err).ToNot(HaveOccurred())
-
-	err = (&atlasstream.AtlasStreamsInstanceReconciler{
-		Client:                      k8sManager.GetClient(),
-		Log:                         logger.Named("controllers").Named("AtlasStreamInstance").Sugar(),
-		GlobalPredicates:            globalPredicates,
-		EventRecorder:               k8sManager.GetEventRecorderFor("AtlasStreamInstance"),
-		AtlasProvider:               atlasProvider,
-		ObjectDeletionProtection:    deletionProtection,
-		SubObjectDeletionProtection: false,
-	}).SetupWithManager(ctx, k8sManager)
-	Expect(err).ToNot(HaveOccurred())
-
-	err = (&atlasstream.AtlasStreamsConnectionReconciler{
-		Client:                      k8sManager.GetClient(),
-		Log:                         logger.Named("controllers").Named("AtlasStreamConnection").Sugar(),
-		GlobalPredicates:            globalPredicates,
-		EventRecorder:               k8sManager.GetEventRecorderFor("AtlasStreamConnection"),
-		AtlasProvider:               atlasProvider,
-		ObjectDeletionProtection:    deletionProtection,
-		SubObjectDeletionProtection: false,
-	}).SetupWithManager(ctx, k8sManager)
-	Expect(err).ToNot(HaveOccurred())
-
-	By("Starting controllers")
-
-	ctx, managerCancelFunc = context.WithCancel(context.Background())
-
-	go func() {
-		err = k8sManager.Start(ctx)
+	By("Setting up deployment controller", func() {
+		err = (&atlasdeployment.AtlasDeploymentReconciler{
+			Client:                      k8sManager.GetClient(),
+			Log:                         logger.Named("controllers").Named("AtlasDeployment").Sugar(),
+			GlobalPredicates:            globalPredicates,
+			EventRecorder:               k8sManager.GetEventRecorderFor("AtlasDeployment"),
+			AtlasProvider:               atlasProvider,
+			ObjectDeletionProtection:    deletionProtection,
+			SubObjectDeletionProtection: false,
+		}).SetupWithManager(k8sManager)
 		Expect(err).ToNot(HaveOccurred())
-	}()
+	})
 
-	return &namespace, managerCancelFunc
-}
+	By("Setting up database user controller", func() {
```

Comment:
nit

I would not have refactor these with such care. All this code needs to die soon.

There should be a single `setupAKOManager` under `pkg/` that can be called from **main**, **integration**, **integration wide** and **e2e** tests all the same.

hopefully next time we touch here is to remove code.